src/rhsm/connection.py

```diff
@@ -2055,14 +2055,14 @@ def getServiceLevelList(self, owner_key: str) -> List[str]:
         results = self.conn.request_get(method, description=_("Fetching service levels"))
         return results
 
-    def getEnvironmentList(self, owner_key: str) -> List[dict]:
+    def getEnvironmentList(self, owner_key: str, list_all: bool = False) -> List[dict]:
         """
         List the environments for a particular owner.
 
         Some servers may not support this and will error out. The caller
         can always check with supports_resource("environments").
         """
-        method = "/owners/%s/environments" % self.sanitize(owner_key)
+        method = "/owners/%s/environments?list_all=%r" % (self.sanitize(owner_key), list_all)
```

Comment:
be careful here: we should not try to use bits that are not supported (ie the new `list_all` query parameter)

Reply:
My understanding is we need to list_all for templates, but I updated this to only use this query parameter if typed environments are an available capability. Is that okay?